apps/web/components/dashboard/BulkBookmarksAction.tsx

```diff
@@ -0,0 +1,204 @@
+}) {
+  return (
+    <Dialog open={open} onOpenChange={setOpen}>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>Delete Bookmarks</DialogTitle>
+          <DialogClose />
+        </DialogHeader>
+        <p>Are you sure you want to delete these bookmarks?</p>
+        <DialogFooter>
+          <Button variant="ghost" onClick={() => setOpen(false)}>
+            Cancel
+          </Button>
+          <Button
+            variant="destructive"
+            onClick={() => {
+              deleteBookmarks();
+              setOpen(false);
+            }}
+          >
+            Delete
+          </Button>
+        </DialogFooter>
+      </DialogContent>
+    </Dialog>
+  );
+}
+
+export default function BulkBookmarksAction() {
+  const { selectedBookmarks, isBulkEditEnabled } = bulkActions();
+  const handleBulkEdit = bulkActions((state) => state.handleBulkEdit);
+  const { toast } = useToast();
+  const pathname = usePathname();
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+
+  useEffect(() => {
+    handleBulkEdit(false); // turn off toggle + clear selected bookmarks on mount
+  }, []);
+
+  const onError = () => {
+    toast({
+      variant: "destructive",
+      title: "Something went wrong",
+      description: "There was a problem with your request.",
+    });
+  };
+
+  const deleteBookmarkMutator = useDeleteBookmark({
+    onSuccess: () => {
+      handleBulkEdit(false);
+    },
+    onError,
+  });
+
+  const updateBookmarkMutator = useUpdateBookmark({
+    onSuccess: () => {
+      handleBulkEdit(false);
+    },
+    onError,
+  });
+
+  interface UpdateBookmarkProps {
+    favourited?: boolean;
+    archived?: boolean;
+  }
+
+  const updateBookmarks = ({
+    favourited,
+    archived,
+  }: UpdateBookmarkProps): void => {
+    selectedBookmarks.map((item) => {
+      updateBookmarkMutator.mutate({
+        bookmarkId: item.id,
+        favourited: favourited ?? item.favourited,
+        archived: archived ?? item.archived,
+      });
+    });
+    toast({
+      description: `${selectedBookmarks.length} bookmarks have been updated!`,
+    });
+  };
+
+  const deleteBookmarks = () => {
+    selectedBookmarks.map((item) => {
+      deleteBookmarkMutator.mutate({ bookmarkId: item.id });
+    });
+    toast({
+      description: `${selectedBookmarks.length} bookmarks have been deleted!`,
+    });
+  };
+
+  const alreadyFavourited =
+    (selectedBookmarks.length &&
+      selectedBookmarks.every((item) => item.favourited === true)) ||
+    pathname.includes("favourites");
+
+  const alreadyArchived =
+    (selectedBookmarks.length &&
+      selectedBookmarks.every((item) => item.archived === true)) ||
+    pathname.includes("archive");
+
+  const actionList = [
+    {
+      name: alreadyFavourited ? "Unfavourite" : "Favourite",
+      icon: (
+        <Star
+          color={alreadyFavourited ? "#ebb434" : "#000"}
+          fill={alreadyFavourited ? "#ebb434" : "transparent"}
```

Comment:
The star doesn't appear at all in dark mode.

Reply:
Fixed